src/electrai/entrypoints/train.py

```diff
@@ -1,176 +1,101 @@
+    # Data
+    # -----------------------------
     train_data, test_data = get_data(cfg)
-    train_loader = DataLoader(train_data, batch_size=int(cfg.nbatch), shuffle=True)
-    test_loader = DataLoader(test_data, batch_size=int(cfg.nbatch), shuffle=False)
-
-    model = GeneratorResNet(
-        n_residual_blocks=int(cfg.n_residual_blocks),
-        n_upscale_layers=int(cfg.n_upscale_layers),
-        C=int(cfg.n_channels),
-        K1=int(cfg.kernel_size1),
-        K2=int(cfg.kernel_size2),
-        normalize=not cfg.normalize_label,
-    ).to(cfg.device)
-
-    optimizer = torch.optim.Adam(
-        model.parameters(), lr=float(cfg.lr), weight_decay=float(cfg.weight_decay)
-    )
-
-    # Linear + Cosine scheduler
-    linsch = torch.optim.lr_scheduler.LinearLR(
-        optimizer, start_factor=1e-5, end_factor=1, total_iters=1
+    train_loader = DataLoader(
+        train_data,
+        batch_size=int(cfg.nbatch),
+        shuffle=True,
+        num_workers=cfg.num_workers,
     )
-    cossch = torch.optim.lr_scheduler.CosineAnnealingLR(
-        optimizer, T_max=int(cfg.epochs) - 1
-    )
-    scheduler = torch.optim.lr_scheduler.SequentialLR(
-        optimizer, [linsch, cossch], milestones=[1]
+    test_loader = DataLoader(
+        test_data,
+        batch_size=int(cfg.nbatch),
+        shuffle=False,
+        num_workers=cfg.num_workers,
     )
 
-    loss_fn = charge.NormMAE()
+    # -----------------------------
+    # Model (LightningModule handles architecture + loss + optimizer)
+    # -----------------------------
+    lit_model = LightningGenerator(cfg)
 
-    prev_loss = float("inf")
-    for t in range(int(cfg.epochs)):
-        logger.info("Epoch %d\n%s", t + 1, "-" * 30)
-        train_epoch(
-            train_loader,
-            model,
-            loss_fn,
-            optimizer,
-            cfg.device,
-            t,
-            accum_iter=int(cfg.nbatch),
-        )
-        test_loss = test_epoch(test_loader, model, loss_fn, cfg.device, t)
+    # -----------------------------
+    # Logging and callbacks
+    # -----------------------------
+    wandb_logger = WandbLogger(
+        project=cfg.wb_pname, name=cfg.wb_ename, entity=cfg.entity, config=vars(cfg)
+    )
 
-        if test_loss < prev_loss:
-            save_checkpoint(t, model, optimizer, scheduler, f"{cfg.model_prefix}.pth")
-            prev_loss = test_loss
+    checkpoint_cb = ModelCheckpoint(
+        monitor="val_loss",
+        save_top_k=2,
+        mode="min",
+        filename=f"{cfg.model_prefix}" + "_{epoch:02d}_{val_loss:.6f}",
+        save_last=True,
+    )
 
-        if t % int(cfg.save_every_epochs) == 1:
-            save_checkpoint(
-                t, model, optimizer, scheduler, f"{cfg.model_prefix}_{t}.pth"
-            )
+    lr_monitor = LearningRateMonitor(logging_interval="epoch")
+
+    # -----------------------------
+    # Profiler
+    # -----------------------------
+    profiler = PyTorchProfiler(
```

Comment:
Let's make the profiler optional rather than always setting it up. Profiling adds overhead, so we should only run it when we explicitly want to profile a run.